### PR TITLE
impl `CanonicalSerde` for `Powers`, and `PartialEq` for params and keys

### PR DESCRIPTION
**Pull request by zhenfeizhang**
## Description

- adding `serdes` for `Powers`
- allow the callers to derive `PartialEq` for `UniversalParams` and `VerifierKey`

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Re-reviewed `Files changed` in the Github PR explorer

---

not relevant

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`

### CODE REVIEW

**Comment by weikengchen** [comment]
just one quick question related to the PR: 

so the PartialEq is added for serialization?

**Comment by alxiong** [comment]
No, `PartialEq` is for accompanying auto-derived `Hash` trait, motivated by cases where consistent hashing of the verifying key is needed.

For example, the predicate identifier in zexe: `pid = H(vk)`

File: src/kzg10/data_structures.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use ark_std::{
 
 /// `UniversalParams` are the universal parameters for the KZG10 scheme.
 #[derive(Derivative)]
-#[derivative(Clone(bound = ""), Debug(bound = ""))]
+#[derivative(Clone(bound = ""), Debug(bound = ""), PartialEq)]
```
Comment (Pratyush) [inline src/kzg10/data_structures.rs:14]:
```suggestion
#[derivative(Clone(bound = ""), Debug(bound = ""), PartialEq(bound = ""), Eq(bound = "")]
```

File: src/kzg10/data_structures.rs [hunk 2]
```diff
@@ -169,9 +170,58 @@ impl<E: PairingEngine> Powers<'_, E> {
...
+        Ok(Self {
+            powers_of_g: Cow::Owned(powers_of_g),
+            powers_of_gamma_g: Cow::Owned(powers_of_gamma_g),
+        })
+    }
+}
 /// `VerifierKey` is used to check evaluation proofs for a given commitment.
 #[derive(Derivative)]
-#[derivative(Default(bound = ""), Clone(bound = ""), Debug(bound = ""))]
+#[derivative(Default(bound = ""), Clone(bound = ""), Debug(bound = ""), PartialEq)]
```
Comment (Pratyush) [inline src/kzg10/data_structures.rs:224]:
```suggestion
#[derivative(Default(bound = ""), Clone(bound = ""), Debug(bound = ""), PartialEq(bound = ""), Eq(bound = "")]
```

**Review by Pratyush — COMMENTED** [review]
LGTM modulo these small changes

**Comment by zhenfeizhang** [comment]
Thanks, and fixed.

**Comment by Pratyush** [comment]
btw, for the hashing, you should note that the `Hash` trait in `std` is not for cryptographic hashing.

**Review by Pratyush — APPROVED** [review]

**Pull request merged**